### PR TITLE
[JENKINS-74890] Extract inline JavaScript from `LogParserWriter.java`

### PR DESCRIPTION
**Pull request by yaroslavafenkin**
## :memo: Description

Fixes CSP violations making plugin CSP compliant.

### :dart: Relevant issues

https://issues.jenkins.io/browse/JENKINS-74890

### :gem: Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js

``` 

## :vertical_traffic_light: How Has This Been Tested?

Manually tested

[Before the change](https://www.youtube.com/watch?v=kCYpefeQvUw)
[After the change](https://www.youtube.com/watch?v=8NpHpVjYRNs)

## :checkered_flag: Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **New and existing unit tests pass locally with my changes**
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/main/resources/hudson/plugins/logparser/LogParserAction/index.jelly [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <j:set var="parseSucceeded" value="${it.result.failedToParseError == null}"/>
   <j:set var="parseFailed" value="${it.result.failedToParseError != null}"/>
   <j:set var="badParseRules" value="${it.result.badParsingRulesError != null}"/>
-  <l:layout>
+  <l:layout title="Parsed Console Output">
```
Comment (yaroslavafenkin) [inline src/main/resources/hudson/plugins/logparser/LogParserAction/index.jelly:7]: This is unrelated to CSP, but title is required on `layout` tag so I added it. Can revert if requested.

File: src/main/java/hudson/plugins/logparser/LogParserConsts.java [hunk 2]
```diff
@@ -25,12 +26,12 @@ public class LogParserConsts {
 
     public static String getHtmlOpeningTags() {
         final String hudsonRoot = Jenkins.get().getRootUrl();
+        final String pluginResourceUrl = String.format("%s/plugin/log-parser/", Jenkins.RESOURCE_PATH).substring(1);
+
         return "<!DOCTYPE html>\n" + "<html>\n" + "\t<head>\n"
                 + "\t\t<title>log-parser plugin page</title>\n"
-                + "\t\t<link type=\"text/css\" rel=\"stylesheet\" href=\""
```
Comment (yaroslavafenkin) [inline src/main/java/hudson/plugins/logparser/LogParserConsts.java:30]: Unsure what those styles were supposed to be, but those don't exist in this repo, nor in core. 404 is received as a response.

File: src/main/resources/hudson/plugins/logparser/LogParserAction/index.jelly (hunk 1 above)
Comment (hypery2k) [reply src/main/resources/hudson/plugins/logparser/LogParserAction/index.jelly:7]: thx for the MR, fine for me

File: src/main/java/hudson/plugins/logparser/LogParserConsts.java (hunk 2 above)
Comment (hypery2k) [reply src/main/java/hudson/plugins/logparser/LogParserConsts.java:30]: thanks for cleaning up

**Comment by basil** [comment]
Thanks @hypery2k! Could this please be released so that we can resolve https://github.com/jenkinsci/acceptance-test-harness/issues/1841?

**Comment by hypery2k** [comment]
sorry, forgotten to release. 2.3.6 is now released: https://github.com/jenkinsci/log-parser-plugin/releases/tag/v2.3.6

**Comment by basil** [comment]
@hypery2k I see the Git tag, but I do not see the new binary uploaded to Artifactory:

https://repo.jenkins-ci.org/artifactory/releases/org/jenkins-ci/plugins/log-parser/

**Comment by hypery2k** [comment]
now it's there

**Comment by basil** [comment]
Thank you! I confirmed that ATH is now passing with CSP enabled. :tada:

**Pull request merged**